src/components/AssetsTableContent.jsx

```diff
@@ -85,7 +85,8 @@ const AssetsTableContent = (props) => {
           {
             props.activePageAssets.map(asset => (
               <TableRowComponent
-                handleViewClick={() => props.handleViewAsset(asset.serial_number)}
+                {...props}
+                viewDetailsRoute={`assets/${asset.serial_number}/view`}
```

Comment:
Why not construct the the URL before the `return` statement and pass it as a string prop? This way, you' ll only have to check for an empty string in the TableRowComponent. I.e.
in AssetsTableContent:
```
const assetViewUrl = `assets/${asset.serial_number}/view`;
return (
  ...
  ...
  <TableRowComponent
    viewDetailsRoute={assetViewUrl}
    ...
  />
)
```
Then in TableRowComponent
```
...
handleView = () => {
  const { viewDetailsRoute, history } = this.props;
  if (!viewDetailsRoute) {
    return null;
  }
  return history.push(viewDetailsRoute);
}
```